eliot-service/eliot/cache_manager.py

```diff
@@ -241,50 +226,113 @@ def _event_generator(self, nonblocking=False):
         else:
             timeout = 1000
 
-        inotify = inotify_simple.INotify(nonblocking=nonblocking)
-        watch_flags = (
-            flags.ACCESS
-            | flags.CREATE
-            | flags.DELETE
-            | flags.MODIFY
-            | flags.MOVED_FROM
-            | flags.MOVED_TO
-        )
-        watch_descriptor = inotify.add_watch(str(self.cachedir), watch_flags)
+        self.inotify = INotify(nonblocking=nonblocking)
+
+        # Set up watches and LRU with what exists already
+        self.watches = OneToOne()
+        self.lru = OrderedDict()
```

Comment:
Maybe it's worth subclassing OrderedDict to an LRU with a clearer API. Something like:

```
# Add or update an item in the LRU
self.lru[path] = size

# Update an item in the LRU
self.lru.touch(path)  # or something like that

# Delete an item from the LRU
del lru[path]
```

Right now we're using `move_to_end` which is an implementation detail and in one case when we update the size, we have to update the size and call `move_to_end`, but updating the size should move it to the end.

I'll think about doing this.